src/components/PupilComponents/QuizEngineProvider/QuizEngineProvider.tsx

```diff
@@ -91,6 +94,25 @@ export const QuizEngineProvider = memo((props: QuizEngineProps) => {
 
   const updateCurrentQuestion = useCallback(
     (incomingQuestionState: Partial<QuestionState>) => {
+      if (
```

Comment:
this is the correct place for this but the name updateCurrentQuestion is a bit confusing now. We aren't changing the current question just its state.

Reply:
for sure